ci/continuous-builder.sh

```diff
@@ -164,6 +167,26 @@ while true; do
       # so we ignore the result code for now
       $TIMEOUT_CMD set-github-status -c ${STATUS_REF} -s $CHECK_NAME/success || $TIMEOUT_CMD report-analytics exception --desc "set-github-status fail on build success"
     fi
+
+    # Look for any code coverage file for the given pull request
+    COVERAGE_SOURCES=$PWD/${PR_REPO_CHECKOUT:-$(basename $PR_REPO)}
+
+    # Push the coverage information to codecov
+    COVERAGE_INFO_DIR=$(find sw/BUILD/ -maxdepth 4 -name coverage.info | head -1 | xargs dirname)
+    if [[ ${COVERAGE_INFO_DIR} ]]; then
+      pushd ${COVERAGE_INFO_DIR}
+        COVERAGE_COMMIT_HASH=${ALIBUILD_HEAD_HASH}
+        if [[ $COVERAGE_COMMIT_HASH == 0 ]]; then
+          COVERAGE_COMMIT_HASH=${ALIBUILD_BASE_HASH}
+        fi
+        $TIMEOUT_CMD bash <(curl -s https://codecov.io/bash) -y .codecov.yml                  \
```

Comment:
Use `curl --max-time` to set a reasonable timeout too (`$TIMEOUT_CMD bash` is evaluated at a later stage, so we need to make `curl` timeout first)